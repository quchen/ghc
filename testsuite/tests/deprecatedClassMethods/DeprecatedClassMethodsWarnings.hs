module DeprecatedClassMethodsWarnings where

data Proxy a = Proxy

instance Functor Proxy where
    fmap = undefined

instance Applicative Proxy where
    pure = undefined
    (<*>) = undefined

instance Monad Proxy where
    (>>=) = undefined

    return = undefined -- should emit a warning
    (>>) = undefined -- should emit a warning
