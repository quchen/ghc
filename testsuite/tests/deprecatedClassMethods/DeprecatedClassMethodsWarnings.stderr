
DeprecatedClassMethodsWarnings.hs:15:5: warning:
    ‘return’ is deprecated, and will be removed in a future release.
    Implement pure from Applicative instead.

DeprecatedClassMethodsWarnings.hs:16:5: warning:
    ‘>>’ is deprecated, and will be removed in a future release.
    Implement (*>) from Applicative instead.
